File: csharp/1017.convert-to-base-2.cs
/*
 * @lc app=leetcode id=1017 lang=csharp
 *
 * [1017] Convert to Base -2
 */

using System.Collections.Generic;
using System.Linq;

// @lc code=start
public class Solution {
    public string BaseNeg2(int N) {
        if (N == 0) {
            return "0";
        }
        var ret = new LinkedList<char>();
        while (N != 0) {
            ret.AddFirst((char)((int)'0' + (N & 1)));
            N = -(N >> 1);
        }
        return new string(ret.ToArray());
    }
}
// @lc code=end

File: java/1017.convert-to-base-2.java
/*
 * @lc app=leetcode id=1017 lang=java
 *
 * [1017] Convert to Base -2
 */

// @lc code=start
class Solution {
    public String baseNeg2(int n) {
        if (n == 0) {
            return "0";
        }
        StringBuilder sb = new StringBuilder();
        while (n != 0) {
            sb.insert(0, n & 1);
            n = -(n >> 1);
        }
        return sb.toString();
    }
}
// @lc code=end

